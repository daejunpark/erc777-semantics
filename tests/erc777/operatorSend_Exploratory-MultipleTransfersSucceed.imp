test.preamble();

// Exploratory testing of operatorSend(From,To,Value, "", "").
// All conditions are met, so it is expected to pass.

test.makeCaller(9);           // switch to 9
test.operatorSend(7, 5, 20, "", "");
test.operatorSend(7, 6, 20, "", "");
test.operatorSend(7, 7, 20, "", "");
test.operatorSend(7, 9, 30, "", "");
