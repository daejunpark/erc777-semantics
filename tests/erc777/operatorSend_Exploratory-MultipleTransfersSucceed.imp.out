Changing the caller to 9
9 is sending 20 from 7 to 5 (data:""),  (operatorData:"")
Balance of 7 is 50
Balance of 5 is 70
9 is sending 20 from 7 to 6 (data:""),  (operatorData:"")
Balance of 7 is 30
Balance of 6 is 80
9 is sending 20 from 7 to 7 (data:""),  (operatorData:"")
Balance of 7 is 30
Balance of 7 is 30
9 is sending 30 from 7 to 9 (data:""),  (operatorData:"")
Balance of 7 is 0
Balance of 9 is 120
